webui/react/src/components/WorkspaceCreateModal.tsx

```diff
@@ -40,14 +48,61 @@ interface Props {
 
 const CodeEditor = React.lazy(() => import('hew/CodeEditor'));
 
+const isNonK8RMError = (e: unknown): boolean => {
+  return e instanceof DetError && e.sourceErr instanceof Response && e.sourceErr['status'] === 404;
+};
+
 const WorkspaceCreateModalComponent: React.FC<Props> = ({ onClose, workspaceId }: Props = {}) => {
   const idPrefix = useId();
-  const { canModifyWorkspaceAgentUserGroup, canModifyWorkspaceCheckpointStorage } =
-    usePermissions();
+  const {
+    canModifyWorkspaceAgentUserGroup,
+    canModifyWorkspaceCheckpointStorage,
+    canSetWorkspaceNamespaceBindings,
+  } = usePermissions();
   const [form] = Form.useForm<FormInputs>();
   const useAgentUser = Form.useWatch('useAgentUser', form);
   const useAgentGroup = Form.useWatch('useAgentGroup', form);
   const useCheckpointStorage = Form.useWatch('useCheckpointStorage', form);
+  const resourceManagers = useAsync(async (canceller) => {
+    try {
+      const response = await getKubernetesResourceManagers(undefined, { signal: canceller.signal });
+      return response.names;
+    } catch (e) {
+      handleError(e, {
+        level: ErrorLevel.Error,
+        publicMessage: 'Failed to fetch Resource Managers.',
+        silent: false,
+        type: ErrorType.Server,
+      });
+      return NotLoaded;
+    }
+  }, []);
+
+  const namespaceBindingsList = useAsync(
+    async (canceller) => {
+      if (workspaceId === undefined) {
+        return {};
```

Comment:
do we wanna return `{}` or `NotLoaded` here?

Reply:
If theres a reason NotLoaded is better, I can change it to that